bot/seasons/evergreen/bookmark.py

```diff
@@ -45,19 +49,54 @@ def __init__(self, bot: commands.Bot):
         embed.set_author(name=target_message.author, icon_url=target_message.author.avatar_url)
         embed.set_thumbnail(url=bookmark_icon_url)
 
+        error_embed = discord.Embed(
+            title=random.choice(ERROR_REPLIES),
+            description=f"Please enable your DMs to receive the bookmark.",
+            colour=Colours.soft_red
+        )
+
         try:
             await ctx.author.send(embed=embed)
         except discord.Forbidden:
-            error_embed = discord.Embed(
-                title=random.choice(ERROR_REPLIES),
-                description=f"{ctx.author.mention}, please enable your DMs to receive the bookmark",
-                colour=Colours.soft_red
-            )
             await ctx.send(embed=error_embed)
         else:
-            log.info(f"{ctx.author} bookmarked {target_message.jump_url} with title '{title}'")
             await ctx.message.add_reaction(Emojis.envelope)
 
+        log.info(f"{ctx.author} bookmarked {target_message.jump_url} with title '{title}'.")
+        await ctx.message.add_reaction(Emojis.pin)
+
+        embed.add_field(
+            name=f'Bookmarked from {ctx.author.name}.',
+            value=f'[Visit original message]({ctx.message.jump_url})',
+            inline=False
+        )
+
+        def check(reaction: discord.Reaction, user: discord.User) -> bool:
+            return (
+                user != self.bot.user
+                and reaction.emoji == Emojis.pin
+                and reaction.message == ctx.message
+                and user not in sent_person
+            )
+
+        sent_person = {ctx.author}  # set of id who got the message

```

Comment:
There is a rare case when `ctx.author` first disabled their DM, but the bot still accepts reaction to add bookmarks for others, the `ctx.author` cannot do it even after turning on their DM because by default the set `sent_person` already have them. There should be a check here if the first DM went through or not.

Reply:
I did that on purpose , As you can see it don't add the envelope reaction but only the pin emoji so that other can get the message it self even if it failed . cuz why not ?

Reply:
and yeah i am making some changes so that person now not in that set.

Reply:
Be careful when you handle it, if it has already bookmarked for the `ctx.author` you do not want that author to react and got another copy of the bookmark.

Reply:
Done :)